File: visa-dev-samplecode-master/vdp-java/src/test/java/com/visa/vdp/vta/TestManageCustomers.java
package com.visa.vdp.vta;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;
import java.util.TimeZone;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestManageCustomers {

    String createCustomerRequest;
    VisaAPIClient visaAPIClient;

    @Test(groups = "vta")
    public void setUp() {
        this.visaAPIClient = new VisaAPIClient();
        SimpleDateFormat sdfDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
        TimeZone utc = TimeZone.getTimeZone("UTC");
        sdfDate.setTimeZone(utc);
        Date now = new Date();
        String strDate = sdfDate.format(now);
        this.createCustomerRequest = "{"
                         + "\"customer\": {"
                                         + "\"cards\": ["
                                             + "{"
                                                 + "\"address\":" + VisaProperties.getProperty(Property.VTA_CREATE_CARD_ADDRESS) + ","
                                                 + "\"billCycleDay\": \"7\","
                                                 + "\"bin\": 431263,"
                                                 + "\"cardEnrollmentDate\": \""+ strDate + "\","
                                                 + "\"cardExpiryDate\": \""+ VisaProperties.getProperty(Property.VTA_CREATE_CUSTOMER_EXPR_DATE)+ "\","
                                                 + "\"cardNickName\": \"My Card\","
                                                 + "\"cardNumber\": \""+ VisaProperties.getProperty(Property.VTA_CREATE_CUSTOMER_CARD_NUM)+ "\","
                                                 + "\"cardSecurityCode\": \"" + VisaProperties.getProperty(Property.VTA_CREATE_CARD_SECURITY_CODE) + "\","
                                                 + "\"contactServiceOfferings\": ["
                                                     + "{"
                                                         + "\"contact\": {"
                                                             + "\"contactNickName\": \"testEmail\","
                                                             + "\"contactType\": \"Email\","
                                                             + "\"contactValue\": \"john@visa.com\","
                                                             + "\"isVerified\": true,"
                                                             + "\"lastUpdateDateTime\": \"" + strDate + "\","
                                                             + "\"mobileCountryCode\": null,"
                                                             + "\"mobileVerificationCode\": null,"
                                                             + "\"mobileVerificationCodeDate\": \"" + strDate + "\","
                                                             + "\"platform\": \"None\","
                                                             + "\"preferredEmailFormat\": \"Html\","
                                                             + "\"securityPhrase\": null,"
                                                             + "\"status\": \"Active\""
                                                         + "},"
                                                         + "\"serviceOfferings\": ["
                                                             + "{"
                                                                 + "\"isActive\": true,"
                                                                 + "\"offeringId\": \"Threshold\","
                                                                 + "\"offeringProperties\": ["
                                                                     + "{"
                                                                         + "\"key\": \"ThresholdAmount\","
                                                                         + "\"value\": \"10\""
                                                                     + "}"
                                                                 + "]"
                                                             + "},"
                                                             + "{"
                                                                 + "\"isActive\": true,"
                                                                 + "\"offeringId\": \"CrossBorder\","
                                                                 + "\"offeringProperties\": ["
                                                                     + "{"
                                                                         + "\"key\": \"ThresholdAmount\","
                                                                         + "\"value\": \"10\""
                                                                     + "}"
                                                                 + "]"
                                                             + "},"
                                                             + "{"
                                                                 + "\"isActive\": true,"
                                                                 + "\"offeringId\": \"Declined\","
                                                                 + "\"offeringProperties\": ["
                                                                     + "{"
                                                                         + "\"key\": \"ThresholdAmount\","
                                                                         + "\"value\": \"10\""
                                                                     + "}"
                                                                 + "]"
                                                             + "},"
                                                             + "{"
                                                                 + "\"isActive\": true,"
                                                                 + "\"offeringId\": \"CardNotPresent\","
                                                                 + "\"offeringProperties\": ["
                                                                     + "{"
                                                                         + "\"key\": \"ThresholdAmount\","
                                                                         + "\"value\": \"10\""
                                                                     + "}"
                                                                 + "]"
                                                             + "}"
                                                         + "]"
                                                     + "}"
                                                 + "],"
                                                 + "\"isActive\": true,"
                                                 + "\"lastFour\": \"" + VisaProperties.getProperty(Property.VTA_CREATE_CUSTOMER_LAST4)  + "\","
                                                 + "\"nameOnCard\": \"Migration\","
                                                 + "\"paused\": false,"
                                                 + "\"portfolioNum\": \"" + VisaProperties.getProperty(Property.VTA_PORTFOLIO_NUMER) + "\","
                                                 + "\"previousCardNumber\": null,"
                                                 + "\"productId\": null,"
                                                 + "\"productIdDescription\": \"Credit\","
                                                 + "\"productType\": \"Credit\","
                                                 + "\"productTypeExtendedCode\": \"Credit\","
                                                 + "\"rpin\": null"
                                             + "}"
                                         + "],"
                                         + "\"communityCode\": \""+ VisaProperties.getProperty(Property.VTA_COMMUNITY_CODE) +"\","
                                         + "\"contacts\": ["
                                             + "{"
                                                 + "\"contactNickName\": \"testEmail\","
                                                 + "\"contactType\": \"Email\","
                                                 + "\"contactValue\": \"john@visa.com\","
                                                 + "\"isVerified\": true,"
                                                 + "\"lastUpdateDateTime\": \"" + strDate + "\","
                                                 + "\"mobileCountryCode\": null,"
                                                 + "\"mobileVerificationCode\": null,"
                                                 + "\"mobileVerificationCodeDate\": \"" + strDate + "\","
                                                 + "\"platform\": \"None\","
                                                 + "\"preferredEmailFormat\": \"Html\","
                                                 + "\"securityPhrase\": null,"
                                                 + "\"status\": \"Active\""
                                             + "}"
                                         + "],"
                                         + "\"customerEnrollmentDate\": \"" + strDate + "\","
                                         + "\"customerId\": \"a1bb6fe1-ea64-4269-b29d-169aebd8780a\","
                                         + "\"firstName\": \"James\","
                                         + "\"isActive\": "+ VisaProperties.getProperty(Property.VTA_CREATE_CUSTOMER_ISACTIVE) +","
                                         + "\"lastName\": \"Bond\","
                                         + "\"preferredCountryCode\": \"" + VisaProperties.getProperty(Property.VTA_PREF_CNTRY_CODE) + "\","
                                         + "\"preferredCurrencyCode\": \"" + VisaProperties.getProperty(Property.VTA_PREF_CRNCY_CODE) + "\","
                                         + "\"preferredFuelAmount\": \"75\","
                                         + "\"preferredLanguage\": \"" + VisaProperties.getProperty(Property.VTA_PREF_LANG) + "\","
                                         + "\"preferredTimeZone\": \"" + VisaProperties.getProperty(Property.VTA_PREF_TIMEZONE) + "\","
                                         + "\"preferredTipPercentage\": \"15\""
                                     + "}"
                                 + "}";
     }

     @Test(groups = "vta")
     public void testGetCustomerDetails() throws Exception {
         String baseUri = "vta/";
         String resourcePath = "v3/communities/"+VisaProperties.getProperty(Property.VTA_COMMUNITY_CODE) +"/customers";

         Map<String,String> headers = new HashMap<String,String>();
         headers.put("ServiceId", VisaProperties.getProperty(Property.VTA_SERVICE_ID));
         CloseableHttpResponse response = this.visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "Create Customer Test", this.createCustomerRequest, MethodTypes.POST, headers);
         Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_CREATED);
         response.close();
     }
}

File: visa-dev-samplecode-master/vdp-python/visa/test/vta/test_manage_customers.py
from visa.helpers.visa_api_client import VisaAPIClient
import json
import datetime
import unittest
import sys
import os
if sys.version_info < (3, 0):
    import ConfigParser as parser
else:
    import configparser as parser
'''
@author: visa
'''

class TestManageCustomers(unittest.TestCase):
    
    config = parser.ConfigParser()
    config_path = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),'..','configuration.ini'))
    config.read(config_path)
    
    def setUp(self):
        date = datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
        self.visa_api_client = VisaAPIClient()
        self.create_customer_request = json.loads('''{
    "customer": {
        "cards": [
            {
                "address": '''+ self.config.get('VDP','vtaCreateCustomerAddress') + ''',
                "billCycleDay": "7",
                "bin": 431263,
                "cardEnrollmentDate": "''' + date + '''",
                "cardExpiryDate": "''' + self.config.get('VDP','vtaCreateCustomerCardExpiryDate') + '''",
                "cardNickName": "My Card",
                "cardNumber": "''' + self.config.get('VDP','vtaCreateCustomerCardNumber') + '''",
                "cardSecurityCode": "''' + self.config.get('VDP','vtaCreateCustomerCardSecurityCode') + '''",
                "contactServiceOfferings": [
                    {
                        "contact": {
                            "contactNickName": "testEmail",
                            "contactType": "Email",
                            "contactValue": "john@visa.com",
                            "isVerified": true,
                            "lastUpdateDateTime": "''' + date + '''",
                            "mobileCountryCode": null,
                            "mobileVerificationCode": null,
                            "mobileVerificationCodeDate": "''' + datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S") + '''",
                            "platform": "None",
                            "preferredEmailFormat": "Html",
                            "securityPhrase": null,
                            "status": "Active"
                        },
                        "serviceOfferings": [
                            {
                                "isActive": true,
                                "offeringId": "Threshold",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            },
                            {
                                "isActive": true,
                                "offeringId": "CrossBorder",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            },
                            {
                                "isActive": true,
                                "offeringId": "Declined",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            },
                            {
                                "isActive": true,
                                "offeringId": "CardNotPresent",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            }
                        ]
                    }
                ],
                "isActive": true,
                "lastFour": "''' + self.config.get('VDP','vtaCreateCustomerLastFour') + '''",
                "nameOnCard": "Migration",
                "paused": false,
                "portfolioNum": "''' + self.config.get('VDP','vtaPortfolioNumber') + '''",
                "previousCardNumber": null,
                "productId": null,
                "productIdDescription": "Credit",
                "productType": "Credit",
                "productTypeExtendedCode": "Credit",
                "rpin": null
            }
        ],
        "communityCode": "''' + self.config.get('VDP','vtaCommunityCode') + '''",
        "contacts": [
            {
                "contactNickName": "testEmail",
                "contactType": "Email",
                "contactValue": "john@visa.com",
                "isVerified": true,
                "lastUpdateDateTime": "''' + date + '''",
                "mobileCountryCode": null,
                "mobileVerificationCode": null,
                "mobileVerificationCodeDate": "''' + datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S") + '''",
                "platform": "None",
                "preferredEmailFormat": "Html",
                "securityPhrase": null,
                "status": "Active"
            }
        ],
        "customerEnrollmentDate": "''' + date + '''",
        "customerId": "a1bb6fe1-ea64-4269-b29d-169aebd8780a",
        "firstName": "James",
        "isActive": ''' + self.config.get('VDP','vtaCreateCustomerIsActive') + ''',
        "lastName": "Bond",
        "preferredCountryCode": "''' + self.config.get('VDP','vtaCreateCustomerPreferedCountryCode') + '''",
        "preferredCurrencyCode": "''' + self.config.get('VDP','vtaCreateCustomerPreferedCurrencyCode') + '''",
        "preferredFuelAmount": "75",
        "preferredLanguage": "''' + self.config.get('VDP','vtaCreateCustomerPreferedLanguage') + '''",
        "preferredTimeZone": "''' + self.config.get('VDP','vtaCreateCustomerPreferedTimeZone') + '''",
        "preferredTipPercentage": "15"
    }
}''')
    
    def test_get_customers(self):
        base_uri = 'vta/'
        resource_path = 'v3/communities/'+ self.config.get('VDP','vtaCommunityCode') +"/customers";
        response = self.visa_api_client.do_mutual_auth_request(base_uri + resource_path, self.create_customer_request, 'Create Customers Test', 'post', {'ServiceId' : self.config.get('VDP','vtaServiceId')})
        self.assertEqual(str(response.status_code) ,"201" ,"Create Customers Test failed")
        pass
